testapp/tests.py

```diff
@@ -60,86 +62,187 @@ def test_create_change_request(self, mock_get_snow_group_guid, mock_pysnow):
             'state': '2'
         }
-        self.mock_pysnow_client.insert.return_value = fake_insert_retval
+
+        fake_resource = mock.MagicMock()
+        fake_resource.create.return_value = fake_insert_retval
+
+        self.mock_pysnow_client.resource.return_value = fake_resource
         mock_pysnow.Client.return_value = self.mock_pysnow_client
 
-        co = self.change_request_handler.create_change_request(
-            'title', 'description', assignment_group='assignment_group'
-        )
+        co = self.change_request_handler.create_change_request('Title', 'Description', payload={})
         last_co = ChangeRequest.objects.last()
 
-        mock_get_snow_group_guid.assert_called_with('assignment_group')
         self.assertEqual(co.pk, last_co.pk)
         self.assertEqual(co.sys_id, fake_insert_retval['sys_id'])
         self.assertEqual(co.number, fake_insert_retval['number'])
         self.assertEqual(co.title, fake_insert_retval['short_description'])
         self.assertEqual(co.description, fake_insert_retval['description'])
         self.assertEqual(co.assignment_group_guid, fake_insert_retval['assignment_group']['value'])
 
-    @mock.patch('django_snow.helpers.snow_request_handler.ChangeRequestHandler._get_snow_group_guid')
-    def test_create_change_request_raises_exception_when_error_in_result(self, mock_get_snow_group_guid, mock_pysnow):
+    def test_create_change_request_parameters(self, mock_pysnow):
+        expected_payload = OrderedDict()
+        expected_payload['type'] = 'normal'
+        expected_payload['assignment_group'] = 'bar'
+        expected_payload['short_description'] = 'Title'
+        expected_payload['description'] = 'Description'
+
+        fake_insert_retval = {
+            'sys_id': uuid.uuid4(),
+            'number': 'CHG0000001',
+            'short_description': 'Title',
+            'description': 'Description',
+            'assignment_group': {'value': uuid.uuid4()},
+            'state': '2'
+        }
+
+        fake_resource = mock.MagicMock()
+        fake_resource.create.return_value = fake_insert_retval
+        self.mock_pysnow_client.resource.return_value = fake_resource
+        mock_pysnow.Client.return_value = self.mock_pysnow_client
+
+        payload = OrderedDict()
+        payload['type'] = 'normal'
+        payload['assignment_group'] = 'bar'
+        self.change_request_handler.create_change_request('Title', 'Description', None, payload=payload)
+        fake_resource.create.assert_called_with(payload=expected_payload)
+
+    def test_create_change_request_default_parameters(self, mock_pysnow):
+        expected_payload = OrderedDict()
+        expected_payload['short_description'] = 'Title'
+        expected_payload['description'] = 'Description'
+        expected_payload['type'] = 'standard'
+        expected_payload['assignment_group'] = 'bar'
+
+        self.change_request_handler.group_guid_dict['assignment_group'] = 'bar'
```

Comment:
I'm not particularly a fan of this. This seems like a bit of a TDD anti-pattern. With TDD, tests should be written before code, but this line is quite obviously written after the code, because it implies, and requires, an intimate knowledge of the code internals. Is there a different way this can be controlled? Perhaps using `django.test.override_settings`?